ui/src/components/canvas/FlowCanvas.tsx

```diff
@@ -151,6 +172,12 @@ const FlowCanvas = () => {
             <ControlButton title="change density" onClick={updateLayoutDensity}>
               <Maximize />
             </ControlButton>
+            <ControlButton title="undo" onClick={() => undo()}>
+              <Undo />
+            </ControlButton>
+            <ControlButton title="redo" onClick={() => redo()}>
```

Comment:
:memo: very minor nitpick, but I think these `onClick` handlers can be shortened to `onClick={undo}` and `onClick={redo}`.

Reply:
it throws a typescript error without the arrow function 
``` Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'number'.```

Reply:
Ah yeah I missed that it has an optional argument.